src/renderer/components/ColorPalette/ColorPalette.js

```diff
@@ -0,0 +1,164 @@
+// -*- mode: js-jsx -*-
+/* Chrysalis -- Kaleidoscope Command Center
+ * Copyright (C) 2018, 2019  Keyboardio, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/**
+ * This is Reactjs functional component that create palette for change buttons color on keyboard
+ */
+import React, { useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import { withStyles } from "@material-ui/core/styles";
+import Paper from "@material-ui/core/Paper";
+import ColorButtonsArea from "./ColorButtonsArea";
+import PickerColorButton from "./PickerColorButton";
+
+ColorPalette.propTypes = {
+  classes: PropTypes.object.isRequired,
+  onColorSelect: PropTypes.func.isRequired,
+  palette: PropTypes.array.isRequired,
+  onColorPick: PropTypes.func.isRequired,
+  disabled: PropTypes.bool.isRequired,
+  selected: PropTypes.number.isRequired
+};
+
+const styles = theme => ({
+  root: {
+    display: "flex",
+    position: "fixed",
+    left: 0,
+    right: 0,
+    bottom: 0,
+    height: 150,
+    justifyContent: "center",
+    alignItems: "flex-start"
+  },
+  palette: {
+    display: "flex",
+    alignItems: "center",
+    padding: 10,
+    [theme.breakpoints.down("sm")]: {
+      width: 470
+    }
+  }
+});
+
+/**
+ * Use to reduce the amount of code
+ * @param {object} color Object with keys that defining colors using the Red-green-blue-alpha (RGBA) model
+ */
+const setColorTamplate = color => ({
```

Comment:
We have a function with this name in two different places and they do two different things. This one returns an object with `{r, g, b, rgb}` and the other returns a string with a css function `rgb(r, g, b)`. We should be able to reuse this one and where the other is used, just call this and take the `rgb` property from it. This way we could move this function to our `utils` folder and then imported into these components, this way it can be reused throughout the application.